resharper/resharper-unity/src/UnityVersion.cs

```diff
@@ -132,7 +133,16 @@ public static Version Parse(string input)
             Version version = null;
             if (match.Success)
             {
-                var type = Convert.ToInt32(groups["type"].Value+groups["revision"], 16);
+                var type = 0;
+                try
+                {
+                    type = Convert.ToInt32(groups["type"].Value + groups["revision"].Value, 16);
```

Comment:
What is this trying to do? Convert something like "f2" into decimal as though it's a hexadecimal string? This will fail with anything that isn't `a` - `f`, and I know Unity have used `p` for patch versions. 

Do we need to know these revision number/types? If so, we'll need to come up with an encoding scheme, e.g. convert the type character into an ASCII byte and store it in the top half of an int, with the revision in the lower half.

Reply:
fixed